storage/innobase/include/data0type.h

```diff
@@ -543,18 +542,21 @@ struct dtype_t{
 					in bytes */
 
 	/** @return whether this is system field */
-	bool vers_sys_field() const { return prtype & DATA_VERSIONED; }
+	bool vers_sys_field() const
+	{
+		return vers_sys_start() || vers_sys_end();
+	}
```

Comment:
Why was this changed? I would not count on the compilers to convert the changed code to a single bitwise AND instruction, similar to how the existing code was probably translated.

Reply:
Function returned `true` for `row_start`, `row_end` and every ordinary versioned fields. While `sys_field` means just `row_start` and `row_end`. It was incorrect in that sense. I fixed it with readable, instead of hand-optimized code.

Reply:
I think that low-level code should be optimized.
We could surely add `ut_ad()` for documentation or readability, like this:
```c++
bool vers_sys_field() const
{
	bool vers_sys = prtype & DATA_VERSIONED;
	ut_ad(vers_sys == vers_sys_start() || vers_sys_end());
	return vers_sys;
}
```

Reply:
There is different semantics when both bits are set. It means: field is _unversioned_ (i.e. "optimized"). The above suggestion will falsely return **true** in such cases.